tests/unit_tests/rng/device/include/moments.hpp

```diff
@@ -0,0 +1,99 @@
+/*******************************************************************************
+* Copyright 2023 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+/*
+*
+*  Content:
+*       oneapi::mkl::rng::device:: distributions moments test (DPC++ interface)
+*
+*******************************************************************************/
+
+#ifndef _RNG_DEVICE_DISTR_MOMENTS_TEST_HPP_
+#define _RNG_DEVICE_DISTR_MOMENTS_TEST_HPP_
+
+#include <iostream>
+
+#if __has_include(<sycl/sycl.hpp>)
+#include <sycl/sycl.hpp>
+#else
+#include <CL/sycl.hpp>
+#endif
+
+#include "oneapi/mkl/rng.hpp"
+#include "oneapi/mkl/rng/device.hpp"
+
+#include "rng_device_test_common.hpp"
+
+template <class Engine, class Distribution>
+class moments_test {
+public:
+    template <typename Queue>
+    void operator()(Queue queue) {
+        using Type = typename Distribution::result_type;
+        // prepare array for random numbers
+        std::vector<Type> r(N_GEN);
+
+        try {
+            sycl::range<1> range(N_GEN / Engine::vec_size);
+
+            sycl::buffer<Type, 1> buf(r);
+            auto event = queue.submit([&](sycl::handler& cgh) {
+                auto acc = buf.template get_access<sycl::access::mode::write>(cgh);
```

Comment:
please, update to the new syntax:

sycl::accessor acc(buf, cgh, sycl::write_only);

Reply:
Updated